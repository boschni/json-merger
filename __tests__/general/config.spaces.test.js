const jsonMerger = require("../../dist");
const { testConfig } = require("../../__helpers__");

describe("when config.stringify is set to \"pretty\" and config.spaces is set", function () {
  test("to a number between 1 and 10 (ex. 3), it will indent with that many spaces", function () {
    const object = {
      a: {
        b: {
          c: 1
        }
      },
    };

    const result = jsonMerger.mergeObject(
      object,
      testConfig({
        stringify: "pretty",
        spaces: 3
      }),
    );

    expect(result).toMatchSnapshot();
  });

  test("to undefined it should indent with tabs", function () {
    const object = {
      a: {
        $replace: 1,
      },
    };

    const result = jsonMerger.mergeObject(
      object,
      testConfig({
        stringify: "pretty",
        spaces: undefined
      }),
    );

    expect(result).toMatchSnapshot();
  });
});
