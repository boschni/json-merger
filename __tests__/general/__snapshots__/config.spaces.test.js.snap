// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when config.stringify is set to "pretty" and config.spaces is set to a number between 1 and 10 (ex. 3), it will indent with that many spaces 1`] = `
"{
   "a": {
      "b": {
         "c": 1
      }
   }
}"
`;

exports[`when config.stringify is set to "pretty" and config.spaces is set to undefined it should indent with tabs 1`] = `
"{
	"a": 1
}"
`;
